File: C++/1734. Decode XORed Permutation.cpp
class Solution {
public:
    vector<int> decode(vector<int>& encoded) {
        int num0 = 0;
        int num1 = 0;
        int n = encoded.size() + 1;

        for(int i=1;i<=n;++i) {
            num0 ^= i;
        }

        for(int i=1;i<n;i+=2) {
            num1 ^= encoded[i];
        }

        vector<int> arr;

        arr.push_back(num0^num1);

        for(auto x:encoded) {
            arr.push_back(arr.back()^x);
        }

        return arr;

    }
};

File: Python3/1734. Decode XORed Permutation.py
class Solution:
    def decode(self, encoded: List[int]) -> List[int]:
        num0 = 0
        num1 = 0

        for i in range(1, len(encoded) + 2):
            num0 ^= i
        
        for i in range(1, len(encoded) + 1, 2):
            num1 ^= encoded[i]
        
        num0 ^= num1

        arr = [num0]

        for x in encoded:
            arr.append(x ^ arr[-1])
        
        return arr
